### DISCUSSION TITLE
Sync to SD Card / DAP / MP3 Player

### DISCUSSION THREAD

**Discussion by plaquette** [audioling/audioling#14] [Ideas]
Hey,

It would be fantastic to have a feature that allows syncing data to a device, even if it's just one-way synchronization.

The key aspect would be preserving the original folder structure—not dumping all files into a single folder but maintaining the hierarchy as it was originally organized.

An incredible bonus would be the ability to sync playlists, along with the corresponding files in their respective folders. Does that make sense?

Best,
P.

**Comment by Clusters** [comment]
I might not understand the full scope of your requirements, but wouldn't this be more like a server feature than a client feature?
For instance, I can sync playlists to my Garmin fitness watch using the Subsonic protocol. I also use rsync to sync on a file system level from the server running Subsonic.

**Comment by hhenne1** [comment]
one of the main features of DSub (subsonic client for android) for me is the ability to save/cache/sync/download all your favorites (as well as recently played ones, if your cache limit allows it) to your sd card, and an offline mode that allows you to play only the saved files. very useful for saving mobile data in general, and not using any mobile data when i don't want to.
A feature like this would allow me to switch to this exclusively on all platforms, if it ever comes close to being in scope.

**Comment by GlassedSilver** [comment]
Respectfully adding to this that play counts, play time/date-stamps should be synced as well so #20 smart playlists and server statistics remain fully functional.

**edit:**
my wish would mean it's a two-way sync system.

**Comment by jeffvli** [comment]
I just want to be clear that audioling is an **online-first** app, meaning that data from your server is not being synced to a local database.
It might not have been clear since it is utilizing a local http server, but the extent to which it functions is to transform the data into an internal model before serving it to the frontend app. 

I also want to clarify that audioling is being built as a **desktop / browser** app with no plans for a native mobile app. I don't believe it's worth it for me to compete in that space when there are many existing options already, not to mention to added complexity.

The plan was to support a limited offline mode which includes features such as:
- Syncing favorites
- Syncing individual playlists
- Syncing individual albums
- Syncing individual tracks

Things which will not be included:
- Local file support

--- 
> The key aspect would be preserving the original folder structure—not dumping all files into a single folder but maintaining the hierarchy as it was originally organized.
>
>An incredible bonus would be the ability to sync playlists, along with the corresponding files in their respective folders. Does that make >sense?

I will look into it, but the goal of the offline mode is to support network-less listening, rather than being able to perform a full directory sync from the server to client devices.

> Respectfully adding to this that play counts, play time/date-stamps should be synced as well so https://github.com/audioling/audioling/discussions/20 smart playlists and server statistics remain fully functional.

This can be added, since it's simply storing scrobbles with their timestamps while playing offline.
